src/System.Net.Primitives/src/System/Net/IPAddress.cs

```diff
@@ -614,7 +615,13 @@ public override int GetHashCode()
             {
                 if (_hashCode == 0)
                 {
-                    _hashCode = StringComparer.OrdinalIgnoreCase.GetHashCode(ToString());
+                    int hashCode = HashHelpers.Combine(HashHelpers.RandomSeed, unchecked((int)PrivateScopeId));
```

Comment:
I assume the random seed is to try to make hash codes less predictable to avoid DoS attacks. Fine. I do notice that in only one other place we use HashHelpers.Combine do we use the random seed.

Reply:
If we care about DoS attacks then we should do the same on the ipv4 calculation below.

@morganbr how much do we want to use randomized hashing on arbitrary types? Or basically just strings?

Reply:
That's the idea, yes. AFAIK currently there are only 2 usages of HashHelper, one with a random seed and the other without.

Reply:
There are several others https://github.com/dotnet/corefx/search?utf8=%E2%9C%93&q=hashhelpers.combine&type=

Reply:
I mainly looked at `ValueTuple`...
Should I remove the seed?

Reply:
What do folks put these into hashtables for? Is it likely to be something that untrusted clients have influence over? The usual DoS concern is people supplying specially crafted strings.

Either way, we should do the same for ipv4.